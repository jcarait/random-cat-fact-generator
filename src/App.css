* {
  margin: 0;
  padding: 0;
}

body {
  background-color: #D9AFD9;
  background-image: linear-gradient(0deg, #D9AFD9 0%, #97D9E1 100%);
}

.App {
  text-align: center;
  min-height: 100vh;
}

.App-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color:#282c34;
}

.title {
  margin-top: 1em;
  margin-bottom: 8px;
}

.bg-page-not-found {
  background-color: #fcb82f;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.image-container {
  display: grid;
  height: 100%;
  min-height: 100vh;
}

.image-404 {
  max-width: 100%;
  max-height: 100vh;
  margin: auto;
}

.attribution {
  position: absolute;
  bottom: 20px;
  left: 10px;
}

.error-message {
  position: absolute;
  top: 23%;
  left: 15%;
  font-size: 2rem;
  font-weight: 600;
  color:firebrick;
  width: 300px;
}

.cat-image {
  max-width: 250px;
  margin-bottom: 50px;
  margin-left: 85px;
}

.container--cat-with-count {
  display: flex;
  align-items: center;
}

@media screen and (max-width: 700px) {

  .cat-image {
    max-width: 125px;
    transition: all 0.3s ease;
    margin-left: 20px;
    margin-bottom: 20px;
  }
}