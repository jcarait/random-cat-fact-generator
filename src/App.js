import "./App.css";
import useFetch from "./api/useFetch";

function App() {
  const { data, isLoading, error } = useFetch("https://catfact.ninja/fact");

  return (
    <div className="App">
      <header className="App-header">
        <h1 className="title">Random Cat Fact Generator</h1>
        <p data-testid="cat-fact">{isLoading ? "Loading..." : data.fact}</p>
        <p>{error === true ? error : ""}</p>
      </header>
    </div>
  );
}

export default App;
