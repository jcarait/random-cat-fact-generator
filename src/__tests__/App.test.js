import { render, screen, act } from "@testing-library/react";
import App from "../App";

test("renders app with div element without crashing", () => {
  const div = document.createElement("div");
  render(<App />, div);
});
