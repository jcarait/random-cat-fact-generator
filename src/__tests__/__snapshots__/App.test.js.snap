// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="title"
    >
      Random Cat Fact Generator
    </h1>
  </header>
  <div
    className="container"
  >
    <div
      className="container--cat-with-count"
    >
      <img
        alt="a cat raising its paw up"
        className="cat-image"
        src="cat.png"
      />
      <div
        className="count"
        data-testid="count"
      >
        <p
          className="count-text"
        >
          0
        </p>
        <p
          className="count-label"
        >
          Number of cat facts refreshed in this session
        </p>
      </div>
    </div>
    <p
      data-testid="loading"
    >
      <span
        style={
          Object {
            "display": "inherit",
          }
        }
      >
        <span
          style={
            Object {
              "animation": "react-spinners-BeatLoader-beat 0.7s 0s infinite linear",
              "animationFillMode": "both",
              "backgroundColor": "#000000",
              "borderRadius": "100%",
              "display": "inline-block",
              "height": "20px",
              "margin": "2px",
              "width": "20px",
            }
          }
        />
        <span
          style={
            Object {
              "animation": "react-spinners-BeatLoader-beat 0.7s 0.35s infinite linear",
              "animationFillMode": "both",
              "backgroundColor": "#000000",
              "borderRadius": "100%",
              "display": "inline-block",
              "height": "20px",
              "margin": "2px",
              "width": "20px",
            }
          }
        />
        <span
          style={
            Object {
              "animation": "react-spinners-BeatLoader-beat 0.7s 0s infinite linear",
              "animationFillMode": "both",
              "backgroundColor": "#000000",
              "borderRadius": "100%",
              "display": "inline-block",
              "height": "20px",
              "margin": "2px",
              "width": "20px",
            }
          }
        />
      </span>
    </p>
  </div>
  <div
    className="container"
  >
    <div
      className="button"
      onClick={[Function]}
    >
      New cat fact
    </div>
  </div>
</div>
`;
